scaladoc/src/main/scala/pink/cozydev/protosearch/scaladoc/ParseScaladoc.scala

```diff
@@ -0,0 +1,172 @@
+          params.flatMap { case clause: Member.ParamClause =>
+            clause.values.collect {
+              case param: Term.Param
+                  if param.default.isEmpty && !param.mods.exists(_.is[Mod.Implicit]) =>
+                val commentDescription: String = paramsComm
+                  .find(_.startsWith(param.name.value))
+                  .getOrElse(param.name.value)
+                val declaredType: String = param.decltpe.map(_.toString).getOrElse("Unknown Type")
+
+                s"$commentDescription: $declaredType"
+            }
+          }
+        }
+
+        val typeParamsComm = commentTokens.collect {
+          case DocToken(TypeParam, Some(name), Some(desc)) => s"@tparam $name: $desc"
+        }
+
+        val typeParamRegex = """def\s+\w+\[([^\]]+)\]""".r
+
+        val typeParams = typeParamRegex.findFirstMatchIn(defn.syntax) match {
+          case Some(matched) =>
+            matched
+              .group(1)
+              .split(",")
+              .map { tparam =>
+                val trimmedTparam = tparam.trim
+                val commentDescription = typeParamsComm
+                  .find(_.startsWith(s"@tparam $trimmedTparam"))
+                  .getOrElse(trimmedTparam)
+                commentDescription.replace("@tparam" + " " + trimmedTparam, trimmedTparam)
+              }
+              .toList
+
+          case None => List()
+        }
+
+        val annotations = defn.mods.collect { case mod: Mod.Annot =>
+          mod.toString
+        }
+
+        val hyperlinks = urlRegex.findAllMatchIn(rawComment).map(_.matched).toList
+
+        val optionalParams = paramss.flatMap { case Member.ParamClauseGroup(_, params) =>
+          params.flatMap { case clause: Member.ParamClause =>
+            clause.values.collect {
+              case param: Term.Param if param.default.isDefined =>
+                val commentDescription: String = paramsComm
+                  .find(_.startsWith(param.name.value))
+                  .getOrElse(param.name.value)
+                val declaredType: String = param.decltpe.map(_.toString).getOrElse("Unknown Type")
+
+                s"$commentDescription: $declaredType"
+            }
+          }
+        }
+
+        val implicitParams = paramss.flatMap { case Member.ParamClauseGroup(_, params) =>
+          params.flatMap { case clause: Member.ParamClause =>
+            clause.values.collect {
+              case param: Term.Param if param.mods.exists(_.is[Mod.Implicit]) =>
+                val commentDescription: String = paramsComm
+                  .find(_.startsWith(param.name.value))
+                  .getOrElse(param.name.value)
+                val declaredType: String = param.decltpe.map(_.toString).getOrElse("Unknown Type")
+
+                s"$commentDescription: $declaredType"
+            }
+          }
+        }
+
+        val allParams = typeParams ++ params ++ optionalParams ++ implicitParams
```

Comment:
Now that we're combining all the various param types, do we need to separate them out in the first place?
Instead of having `optionalParams` and `implicitParams` could we modify `params` to capture all types by removing the guard clause on the match?
e.g. going from
```
case param: Term.Param
  if param.default.isEmpty && !param.mods.exists(_.is[Mod.Implicit])
```
to just
```
case param: Term.Param
```